File: shared/src/iosMain/kotlin/com/tobiapplications/kmpmeetup/utils/di/Koin.kt
package com.tobiapplications.kmpmeetup.utils.di

import org.koin.core.KoinApplication
import org.koin.core.component.KoinComponent
import org.koin.core.context.startKoin


object Koin : KoinComponent {
    fun initKoin(): KoinApplication {
        return startKoin {
            modules(KoinShared.sharedModules)
        }
    }
}

File: iosApp/iosApp/di/Koin.swift
import shared


func startKoin() {
    let koinApplication = Koin().doInitKoin()
    _koin = koinApplication.koin
}

private var _koin: Koin_coreKoin? = nil
var koin: Koin_coreKoin {
    return _koin!
}
